modules/plugin/coverage-multidim/netcdf/src/test/java/org/geotools/coverage/io/netcdf/NetCDFReaderTest.java

```diff
@@ -195,6 +198,43 @@ public void testFullReadOnCoverageWithIncreasingLat() throws IOException, Factor
         }
     }
 
+    private String testFilePath = "/Users/devon/Downloads/data-ticket/HYCOM/hycom_glb_regp01_t003.nc";
+
+    @Test
+    public void testOffsetAndScale() throws IOException, FactoryException, ParseException {
+        File file = new File(testFilePath);
+        final Hints hints= new Hints();
+        // Get format
+        final AbstractGridFormat format = GridFormatFinder.findFormat(file.toURI().toURL(),hints);
+        final NetCDFReader reader = (NetCDFReader) format.getReader(file.toURI().toURL(), hints);
+
+        String coverageName = "surf_el";
+
+        final ParameterValue<List> time = ImageMosaicFormat.TIME.createValue();
+        final SimpleDateFormat formatD = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm");
+        formatD.setTimeZone(TimeZone.getTimeZone("GMT"));
+        final Date timeD = formatD.parse("2014-07-07T03:00");
+        time.setValue(new ArrayList() {
+            {
+                add(timeD);
+            }
+        });
+
+
+        final ParameterValue<GridGeometry2D> gg = AbstractGridFormat.READ_GRIDGEOMETRY2D
+            .createValue();
+        gg.setValue(new GridGeometry2D(reader.getOriginalGridRange(coverageName), reader.getOriginalEnvelope(coverageName)));
+
+        GeneralParameterValue[] values = new GeneralParameterValue[] {};
+        GridCoverage2D coverage = reader.read(coverageName, values);
+
+        float[] result = coverage
+            .evaluate((DirectPosition) new DirectPosition2D(DefaultGeographicCRS.WGS84, 260.0f, 0.0f),
+                new float[1]);
+
+        assertEquals(0.264f, result[0], 0.0000f);
```

Comment:
Using a tolerance of ```1e-6f``` future-proofs against a JVM implementation that introduces a tiny rounding error when expanding scale and offset.